effects/effect/effects-setup-hook.sh

```diff
@@ -3,6 +3,13 @@
 # ----------------------------------------------------------------------------
 # prepare headers file for curl to talk to Hercules CI
 
+installFSRootFiles() {
+  # Compatibility hook: hercules-ci-agent <= 0.10.5 does not copy for us. Requires rw root.
+  if [[ -z "${__hci_effect_fsroot_copied:-}" && -n "${__hci_effect_fsroot_copy:-}" ]]; then
+    cp --no-preserve=ownership --recursive --reflink=auto -T "$__hci_effect_fsroot_copy"/ /;
+  fi
+}
+preInitHooks+=("installFSRootFiles")
```

Comment:
Next step is to support this natively in `hercules-ci-agent` for the case when `__hci_effect_root_readonly=1`.
It can set `__hci_effect_fsroot_copied` to cleanly disable this compat logic.